app/forms/delivery_partner_form.rb

```diff
@@ -3,20 +3,22 @@
 class DeliveryPartnerForm
   include ActiveModel::Model
 
-  attr_accessor :name, :lead_providers, :provider_relationships
+  attr_accessor :name, :lead_providers, :provider_relationships, :provider_relationship_hashes
   validate :lead_providers_and_cohorts_validation
 
+  def self.provider_relationship_value(lead_provider, cohort)
+    "{\"lead_provider_id\": \"#{lead_provider.id}\", \"cohort_id\": \"#{cohort.id}\"}"
+  end
+
   def available_lead_providers
     LeadProvider.joins(:cohorts).includes(:cohorts).select { |lead_provider| lead_provider.cohorts.any? }
   end
 
   def chosen_provider_relationships
-    provider_relationships.map do |relationship|
-      ProviderRelationship.new(
-        cohort: Cohort.find(relationship["cohort_id"]),
-        lead_provider: LeadProvider.find(relationship["lead_provider_id"]),
-      )
-    end
+    provider_relationship_hashes
+      &.map { |value| JSON.parse(value) }
+      &.filter { |relationship_params| lead_providers.include?(relationship_params["lead_provider_id"]) }
```

Comment:
lead_providers gives the impression that it is a collection of whole objects - alternatively using lead_provider_ids as a name would make it easier to understand the intention